enterprise/server/remote_execution/cgroup/cgroup.go

```diff
@@ -31,6 +32,153 @@ const (
 	cidPlaceholder = "{{.ContainerID}}"
 )
 
+var (
+	// ErrV1NotSupported is returned when a function does not support cgroup V1.
+	ErrV1NotSupported = fmt.Errorf("cgroup v1 is not supported")
+)
+
+// GetCurrent returns the cgroup of which the current process is a member.
+//
+// The returned path is relative to the cgroupfs root, but begins with a slash.
+//
+// For example, if the current process is part of "/sys/fs/cgroup/foo", this
+// returns "/foo".
+func GetCurrent() (string, error) {
+	b, err := os.ReadFile("/proc/self/cgroup")
+	if err != nil {
+		return "", fmt.Errorf("read cgroup from procfs: %w", err)
+	}
+	parts := strings.Split(string(b), ":")
+	if len(parts) < 3 {
+		return "", fmt.Errorf("invalid /proc/self/cgroup value %q", err)
+	}
+	if controllers := parts[1]; controllers != "" {
+		return "", ErrV1NotSupported
+	}
+	return strings.Join(parts[2:], ":"), nil
+}
+
+// Setup configures the cgroup at the given path with the given settings.
+// Any settings for cgroup controllers that aren't enabled are ignored.
+// IO limits are applied to the given block device, if specified.
+func Setup(ctx context.Context, path string, s *scpb.CgroupSettings, blockDevice *block_io.Device) error {
```

Comment:
i think this diff is from your previous PR

Reply:
woops, fixed